fractal_server/app/routes/auth/group.py

```diff
@@ -212,3 +214,27 @@ async def delete_single_group(
     await db.commit()
 
     return Response(status_code=status.HTTP_204_NO_CONTENT)
+
+
+@router_group.patch("/group/{group_id}/user-settings/", status_code=200)
+async def patch_user_settings_bulk(
+    group_id: int,
+    settings_update: UserSettingsUpdate,
+    superuser: UserOAuth = Depends(current_active_superuser),
+    db: AsyncSession = Depends(get_async_db),
+):
+    await _usergroup_or_404(group_id, db)
+    res = await db.execute(
+        select(UserSettings)
+        .join(UserOAuth)
+        .where(LinkUserGroup.user_id == UserOAuth.id)
+        .where(LinkUserGroup.group_id == group_id)
+    )
+    settings_list = res.scalars().all()
+    for settings in settings_list:
+        for k, v in settings_update.dict(exclude_unset=True).items():
```

Comment:
Minor update:

Can we pre-compute `settings_update.dict(exclude_unset=True).items()` before entering the first loop?
Alternative option: let's switch the two `for` loops.